### PR TITLE
Bugfix/issue-174 disable append-only stream left all join append-only stream

### LINKED ISSUE #174: Disable append-only stream left join append-only stream.
I am running a stream/stream left join, and the result seems not what I am expecting
```
SELECT v1.window_start, v1.value, v2.value 
FROM v_iot_10s as v1
LEFT JOIN v_timer_20s as v2
ON v1.window_start=v2.window_start 
AND date_diff_within(1m, v1.window_start, v2.window_start)
```
left stream has a interval 10s, and right stream has a interval 20s, so it should have a zero every 20s since I am joining by time, but it seems not, every 20s, there are two results of join, can someone explain how this join works?
note on 20s  and 40s  there are two results (edited) 

image

### PR DESCRIPTION
**Pull request by lizhou1111**
Disable append-only stream left join append-only stream and block some smoke test about this.So far, the `left join` is only supported in `data enrichment join`.The join results can not be guaranteed.

### CODE REVIEW

File: tests/stream/test_stream_smoke/0018_query_state12_streaming_join.json [hunk 1]
```diff
@@ -273,7 +273,7 @@
         },
         {
             "id": 356,
-            "tags": ["query_state", "data_enrichment_join", "left-join"],
```
Comment (yl-lisen) [inline tests/stream/test_stream_smoke/0018_query_state12_streaming_join.json:276]: modify tag `data_enrichment_join` to `bidirectional join`

File: tests/stream/test_stream_smoke/0009_stream_join_stream_case2.json [hunk 2]
```diff
@@ -190,7 +190,7 @@
         },
         {
             "id": 56,
-            "tags": ["data_enrichment_join", "left-join"],
```
Comment (yl-lisen) [inline tests/stream/test_stream_smoke/0009_stream_join_stream_case2.json:193]: revise tag data_enrichment_join to bidirectional join

File: tests/stream/test_stream_smoke/0020_seek_to.json [hunk 3]
```diff
@@ -353,9 +353,7 @@
...
-            {"client":"python", "wait":2, "query_id":"2101", "depends_on_stream":"test21_limit_by_stream", "query_end_timer":3, "query_type": "stream", "query":"select a.i, count() from test21_limit_by_stream as a join test21_limit_by_stream as b on a.i = b.i where a._tp_time >= '2022-12-01 00:00:01.111' and b._tp_time >= '2022-12-01 00:00:02.111' group by a.i emit periodic 1s"},
-            {"client":"python", "wait":2, "query_id":"2102", "depends_on_stream":"test21_limit_by_stream", "query_end_timer":3, "query_type": "stream", "query":"select i, s from test21_limit_by_stream as a left join test21_limit_by_stream_2 as b on a.i = b.i where test21_limit_by_stream._tp_sn > 1 and default.test21_limit_by_stream_2._tp_sn > 2"},
-            {"client":"python", "wait":2, "query_id":"2103", "depends_on_stream":"test21_limit_by_stream", "query_end_timer":3, "query_type": "stream", "query":"select count() from test21_limit_by_stream as a left join test21_limit_by_stream_2 as b on a.i = b.i where test21_limit_by_stream._tp_sn > 0 and default.test21_limit_by_stream_2._tp_sn < 4 emit periodic 1s"}
```
Comment (yl-lisen) [inline tests/stream/test_stream_smoke/0020_seek_to.json:358]: keep these cases, only change to supported Join

**Comment by yokofly** [comment]
https://github.com/timeplus-io/proton/actions/runs/6555033796/job/17803789771#step:7:5586
UT failure: local can run with 
```
./src/unit_test_dbms
```

```
[ RUN      ] StreamingHashJoin.AppendLeftAllJoinAppend
unknown file: Failure
C++ exception with description "'Append stream' LEFT ALL JOIN 'Append stream' is not supported: 
While running (Test case: <Append> left all JOIN <Append>, ON clause: t1.col_1 = t2.col_1, left header: 'col_1 int32 int32(size = 0), col_2 datetime64(3, 'UTC') datetime64(size = 0)', right header: 'col_1 int32 int32(size = 0), col_2 datetime64(3, 'UTC') datetime64(size = 0)')" thrown in the test body.
[  FAILED  ] StreamingHashJoin.AppendLeftAllJoinAppend (29 ms)
```

**Review by yl-lisen — APPROVED** [review]

**Pull request merged**